### DISCUSSION TITLE
[VS 2022] TAB key does not accept the answer because of conflicting setting

### DISCUSSION THREAD

**Discussion by Mitsuriou** [community/community#15221] [Copilot Conversations] [answered]
Hello,
I'm coding in C++, and I have a lot of issues to accept solution given by Copilot:
- For the raw source code, I cannot accept with TAB nor the popup.
- For the comments, I can only accept with TAB but not with the popup.
I am using VS 2022 with default extensions + "Qt VS Tools", almost default settings, and basic Visual Studio SLN.
I'm on Windows 10 latest update available, Ryzen 7 3700x CPU + 32Gb 3200Mhz RAM.

![copilot_cannot_accept](https://user-images.githubusercontent.com/37184309/164083845-5c8536a5-7b8f-4955-9a87-a331c29db5ee.gif)

**Comment by FilipKufalov** [comment]
Make sure to check if TAB has the following command when pressed
![image](https://user-images.githubusercontent.com/99423097/164085290-e685f163-636d-434f-8ac2-4813a4b501ac.png)

`{
  "key": "tab",
  "command": "editor.emmet.action.expandAbbreviation",
  "when": "config.emmet.triggerExpansionOnTab && editorTextFocus && !editorReadonly && !editorTabMovesFocus"
}`

**Reply by Mitsuriou** [reply]
I'm using Visual Studio 2022 and not Visual Studio Code ;)
Yet, I found the conflicting setting myself ^^

**Reply by FilipKufalov** [reply]
> I'm using Visual Studio 2022 and not Visual Studio Code ;) Yet, I found the conflicting setting myself ^^

I tried to implement Copilot in Visual Studio 2022 but I cant find an Extension, do you have a tutorial/guide on how to do it?

**Reply by Mitsuriou** [reply]
> 

It is quite straight forward, just got to "Extensions" > "Manage extensions" > "Online" < then type "Copilot", download the extensions and restart your VS. Then you will need to click on a login popup in VS, and you are good to go ;)
Please note that the extension is only available for VS 2022 (not available for 2017 / 2019).

**Reply by FilipKufalov** [reply]
I'm updating now VS to 17.1.15 Update arrived 3 hours ago

![image](https://user-images.githubusercontent.com/99423097/164086958-07738c40-c122-40e3-8f42-568e8d2c487a.png)

**Reply by Mitsuriou** [reply]
I am also using the latest version of VS2022.
Maybe you can use this link to the store ? https://marketplace.visualstudio.com/items?itemName=GitHub.copilotvs
Another thing, are you logged in to the account you asked a Copilot BETA trial for?

![image](https://user-images.githubusercontent.com/37184309/164087791-50ba4bda-cd0f-4a59-bbb9-6d383d500514.png)

**Comment by Mitsuriou** [comment, marked as answer]
Just found that this setting (the very first checkbox) caused issues with Copilot:
![image](https://user-images.githubusercontent.com/37184309/164085623-af766701-6825-4a86-93c8-7d2046983f0d.png)

**Reply by ColdIronPoz** [reply]
For those in English: Options -> Text Editor -> C/C++-> Code Style -> Formatting -> Automatically indent when I type a tab
must be unchecked.

![image](https://user-images.githubusercontent.com/95450339/164305414-dbafd033-25be-42d3-9007-4a56533bc8cd.png)

**Comment by rungrotArunon** [comment]
Thanks you

**Comment by knilecrack** [comment]
If someone else has this issue you can try changing keybind in Tools -> Options -> Keyboard -> Edit.InsertTab and bind it to something else.

**Comment by MeroZemory** [comment]
In my case, I solved it by resetting the settings. I used to use the "Visual C++" collection of settings, but I solved it using "General". I'm guessing I messed with the wrong key bindings or tab-related settings, or that the "Visual C++" collection has tab-related key bindings.

**Comment by HamRusTal** [comment]
Related [MS VS Dev Community Issue](https://developercommunity.visualstudio.com/t/copilot-tab-key-to-accept-suggestion-con/10734081) — please feel free to vote.
